packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/sections/components/ReminderSection/RemindersSection.tsx

```diff
@@ -0,0 +1,189 @@
+    };
+
+  const renderRemindersSection = () => {
+    if (
+      isReadPaymentReminderAllowedLoading ||
+      isReadOverdueReminderAllowedLoading ||
+      isPaymentRemindersLoading ||
+      isOverdueRemindersLoading
+    ) {
+      return <Typography>{t(i18n)`Loading...`}</Typography>;
+    }
+
+    if (!areRemindersEnabled) {
+      return (
+        <Alert severity="warning">
+          {t(i18n)`Reminders are disabled for this counterpart.`}
+        </Alert>
+      );
+    }
+
+    return (
+      <>
+        {!isEmailValid && (
+          <Alert severity="warning">
+            {t(
+              i18n
+            )`No default email for selected Counterpart. Reminders will not be sent.`}
+          </Alert>
+        )}
+        <Box mt={2} />
+        <Grid container spacing={3}>
+          <Grid item xs={12} sm={6}>
+            <Controller
+              name="payment_terms_id"
+              control={control}
+              render={({ field, fieldState: { error } }) => (
+                <SelectFieldWithEdit
+                  field={field}
+                  error={error}
+                  label={t(i18n)`Before due date`}
+                  options={paymentReminders?.data || []}
+                  noOptionsText={t(i18n)`No payment reminders available`}
+                  disabled={disabled}
+                  root={root as HTMLElement}
+                  details={paymentIDReminder}
+                  createOptionLabel={t(i18n)`Create a reminder preset`}
+                  handleSelectChange={(event) =>
+                    handleSelectChange(event, 'payment')
+                  }
+                  control={control}
+                />
+              )}
+            />
+          </Grid>
+          <Grid item xs={12} sm={6}>
+            <Controller
+              name={'overdue_reminder_id' as keyof CreateReceivablesFormProps}
+              control={control}
+              render={({ field, fieldState: { error } }) => (
+                <SelectFieldWithEdit
+                  field={field}
+                  error={error}
+                  label={t(i18n)`Overdue reminders`}
+                  options={overdueReminders?.data || []}
+                  noOptionsText={t(i18n)`No overdue reminders available`}
+                  disabled={disabled}
+                  details={overdueIDReminder}
+                  createOptionLabel={t(i18n)`Create a reminder preset`}
+                  root={root as HTMLElement}
+                  handleSelectChange={(event) =>
+                    handleSelectChange(event, 'overdue')
+                  }
+                  control={control}
+                />
+              )}
+            />
+          </Grid>
+        </Grid>
+      </>
+    );
+  };
+
+  return (
+    <Stack spacing={1}>
+      <Typography variant="subtitle2">{t(i18n)`Reminders`}</Typography>
+      <Card variant="outlined" sx={{ borderRadius: 2 }}>
+        <CardContent>{renderRemindersSection()}</CardContent>
```

Comment:
This is definitely the case we discussed in the last time. Render function is an anti-pattern in React for such cases.

Reply:
Changed to avoid render fn pattern